feat: variadic input helpers (#284)
danielroe/magic-regexp: docs/content/2.getting-started/2.usage.md

Every pattern you create with the library should be wrapped in `createRegExp`, which enables the build-time transform.

> `createRegExp` accepts arbitrary number of arguments of type `string`, or `Input` that are built up using helpers from `magic-regexp`, and an optional final argument of an array of flags or flags string. It creates a `MagicRegExp`, which has all the patterns from the inputs concatenated.

Reviewer: ```suggestion
`createRegExp` accepts an arbitrary number of arguments of type `string` or `Input` (built up using helpers from `magic-regexp`), and an optional final argument of an array of flags or a flags string. It creates a `MagicRegExp`, which concatenates all the patterns from the arguments that were passed in.
```

createRegExp('string-to-match', [global, multiline])
// you can also pass flags directly as strings or Sets
createRegExp('string-to-match', ['g', 'm'])

> // or pass in multiple `string` and `input patterns`,
> // all inputs will be concatenated to one RegExp pattern
> createRegExp(
>     'foo',
>     maybe('bar').groupedAs('g1'),
>     'baz',
>     [global, multiline]
> ) 
> // equivalent to /foo(?<g1>(?:bar)?)baz/gm

Reviewer: ```suggestion
// or pass in multiple `string` and `input patterns`,
// all inputs will be concatenated to one RegExp pattern
createRegExp(
  'foo',
  maybe('bar').groupedAs('g1'),
  'baz',
  [global, multiline]
) 
// equivalent to /foo(?<g1>(?:bar)?)baz/gm
```
